### PR TITLE
Pull request for metric-list-attribute-filter

### PR DESCRIPTION
**Pull request by jd**
indexer: replace ids argument with attribute filter in list_metrics()
indexer: replace names filter in list_metric by attribute_filter
indexer: replace kwargs filtering by regular attribute filter

### CODE REVIEW

**Comment by jd** [comment]
Pull-request updated, HEAD is now f934488054169310f2deaa2d577988651a6b13a1

**Comment by jd** [comment]
This is a first step toward fixing https://github.com/gnocchixyz/gnocchi/issues/314

File: gnocchi/tests/test_indexer.py [hunk 1]
```diff
@@ -1121,9 +1124,6 @@ def test_get_metric_with_bad_uuid(self):
         self.assertEqual([], self.index.list_metrics(
             attribute_filter={"=": {"id": e1}}))
 
-    def test_get_metric_empty_list_uuids(self):
```
Comment (chungg) [inline gnocchi/tests/test_indexer.py]: what happens in this scenario now?

Comment (jd) [reply gnocchi/tests/test_indexer.py]: Same things. it's basically `{"in" : {"id": []}}`

it's not a special case anymore that's why I removed the test

Comment (chungg) [reply gnocchi/tests/test_indexer.py]: does it crash? i'm pretty sure it was there to make sure it didn't throw some SQL error which it did previously with empty list

**Comment by jd** [comment]
Pull-request updated, HEAD is now 1de30ae5e448c16d46aa9a453da47858ee76c012

**Comment by jd** [comment]
Pull-request updated, HEAD is now 5395f7481aa1a702b6078b686983bc06bc95342e

**Pull request merged**